index.js

```diff
@@ -836,6 +836,21 @@ class Command extends EventEmitter {
     return this;
   };
 
+  /**
+   * Factory routine to create a new unattached argument.
+   *
+   * See .argument() for creating an attached argument, which uses this routine to
+   * create the argument. You can override createArgument to return a custom argument.
+   *
+   * @param {string} name
+   * @param {string} [description]
+   * @return {Argument} new argument
+   */
+
+  createArgument(name, description) {
+    return new Argument(name, description);
```

Comment:
What are the benefits of adding this factory when you can use `new Argument()` directly?

Reply:
The main purpose is so you can override the method and return a custom subclass of Argument, which will be used to create the arguments added by Command methods `.argument()` and `.arguments()` and `.command('sub <file>`).

This factory pattern was first offered with `.createCommand()` where it is more likely to be used, but offering the same pattern for Option and now Argument.

Reply:
But why will you want to override the createArgument method when you can just pass any custom Argument class to addArgument?
also, in order create custom argument, you need to somewhat know the internals of Argument class (in both the factory solution or by using it directly), so it sounds like a bad practice and a doc bloater, for probably a really rare use case

Reply:
`.createCommand()` was the first one added in #1191 to make Command more subclass friendly. `.createHelp()` was added in major help refactor in #1365.

`.createOption()` was added in #1380 to follow the pattern. We had established a tidy pattern for `Command` which also worked for `Help`, so following through and offering same support for `Option`.

> in order to create custom argument, you need to somewhat know the internals of the Argument class

Yes, that is implicit in writing a subclass.

I did consider whether to directly support subclassing at all for `Command`, but it a useful approach for extending functionality especially in custom cases which are not common use cases and would not be integrated into Commander itself.